ddtrace/contrib/pytest/_plugin_v2.py

```diff
@@ -270,6 +277,7 @@ def _pytest_runtest_protocol_post_yield(item, nextitem, coverage_collector):
         if CISuite.is_item_itr_skippable(suite_id) and not CISuite.was_forced_run(suite_id):
             CISuite.mark_itr_skipped(suite_id)
         else:
+            _handle_coverage_dependencies(suite_id)
             CISuite.finish(suite_id)
         if nextitem is None or (next_test_id is not None and next_test_id.parent_id.parent_id != module_id):
```

Comment:
Not related to this PR, but didn't we have an issue where if the tests are randomized we can't rely on this order? What was the solution to this (or did I misunderstand the problem)?

Reply:
We do, yes, but in this case, it only means that we would do this more expensive computation more times than necessary.

In the new API, the coverage data is accumulated until the suite is `.finish()`ed , and that only happens if all the suite's children are finished, or if the finish is forced (this is something I introduced to be part of the workaround with randomization, though it's not the whole solution itself).

What we could do is introduce something like `CISuite.has_unfinished_children() -> bool` that would return `False`.

I thought about it before but didn't because `.finish()` already does that check for items with children, so it was a little redundant.